v2/pkg/ast/path.go

```diff
@@ -96,26 +97,42 @@ func (p Path) String() string {
 }
 
 func (p Path) DotDelimitedString() string {
-	out := ""
+	builder := strings.Builder{}
```

Comment:
What if you use a bytes buf pool with a pre-defined size (toGrow) or a string.Builder pool that you grow to the required size? Currently, you're creating one buffer per func call.

Reply:
toGrow could not be predefined, it varies depending on a path size

reusing Builder is not possible, it uses unsafe string
https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/strings/builder.go;l=49